### PR TITLE
2517 send transaction timeout

### LINKED ISSUE #2517: SendTransaction Timeout
Sometimes Infura takes a long time to return the transaction, the socket times out but the dialog just hangs. Usually, the transaction has been sent and AW picks up the transaction as normal, in a minute or so.

Either: Add a 30 second timeout for Transaction Sending, which should be done in a non-duplicative way - sendTransaction is called from various places; however it always ends up in one of these methods:

```resendTransaction```
```createTransactionWithSig``` 
```create1559TransactionWithSig```

Or: See if hooking up the timeout from the AWHttpService is possible; it picks up other exceptions and relays them to the exception handler. You might have to experiment with some 'fake' exceptions or generate your own timeouts after a certain time.

### PR DESCRIPTION
**Pull request by asif-finimble**
Closes #2517

### CODE REVIEW

**Review by JamesSmartCell — CHANGES_REQUESTED** [review]
Loos good, I have a few suggestions -

(I think) the timeout should be on the write, not the read, since we're sending a transaction. You can update this to 30 seconds universally, so no need for the extra parameter.

Second - can you catch the exception from the sendtransaction into a separate function that calls either the usual fail or the specific timeout fail - the conditional between two dialogs within the function is less flexible and more prone to breaking in the future.

**Comment by asif-finimble** [comment]
Ok. Will make the changes.   
Also the exception is thrown by viewModel via liveData. So all errors will be received in the ```txError()``` method. I'll separate out the different dialogs (timeout/error) into separate functions.

File: app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java [hunk 1]
```diff
@@ -23,6 +29,7 @@
 import io.reactivex.Single;
 import io.reactivex.schedulers.Schedulers;
 import io.realm.Realm;
+import timber.log.Timber;
 
 public class TransactionRepository implements TransactionRepositoryType {
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java:35]: Can the changes to this file be reverted? There doesn't appear to be any change here.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -926,4 +926,6 @@
     <string name="unable_store_key">Unable to store key: %1$s</string>
     <string name="title_ens_lookup_warning">ENS Lookup Warning</string>
     <string name="message_ens_node_not_sync">AlphaWallet has detected a discrepancy between ethereum\'s timestamp and your device\'s current time, or possibly the blockchain connection is not synchronized, please check your phone\'s time is correct, or ignore this warning.</string>
+    <string name="message_transaction_timeout">The transaction has timed out. You can view the status in activity page.</string>
+    <string name="error_transaction_timeout">Transaction Timeout</string>
 </resources>
```
Comment (JamesSmartCell) [inline app/src/main/res/values/strings.xml:931]: What about the strings for the other languages?

**Review by JamesSmartCell — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (asif-finimble) [reply app/src/main/res/values/strings.xml:931]: Done

File: app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java (hunk 1 above)
Comment (asif-finimble) [reply app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java]: Done

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (JamesSmartCell) [reply app/src/main/res/values/strings.xml:931]: How about -es ?

File: app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java [hunk 3]
```diff
@@ -15,6 +16,11 @@
...
 import org.web3j.protocol.Web3j;
 import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.protocol.core.methods.response.EthTransaction;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpType;
 import org.web3j.utils.Numeric;
 
 import java.math.BigInteger;
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java:26]: Could you do a 'compare with origin/master' and sync to that?

**Review by JamesSmartCell — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java (hunk 3 above)
Comment (asif-finimble) [reply app/src/main/java/com/alphawallet/app/repository/TransactionRepository.java]: Sorry I missed -es.   
Compared and synced TransactionRepository with master

**Review by JamesSmartCell — APPROVED** [review]

**Pull request merged**